src/hotspot/share/opto/arraycopynode.cpp

```diff
@@ -629,6 +641,15 @@ Node *ArrayCopyNode::Ideal(PhaseGVN *phase, bool can_reshape) {
   }
 
   if (!finish_transform(phase, can_reshape, ctl, mem)) {
+    if (can_reshape) {
+      // make sure the just created clones are taken care of if they now are dead
+      if (!forward_mem->is_top() && forward_mem->outcnt() == 0) {
+        phase->is_IterGVN()->_worklist.push(forward_mem);
+      }
+      if (!backward_mem->is_top() && backward_mem->outcnt() == 0) {
+        phase->is_IterGVN()->_worklist.push(backward_mem);
+      }
+    }
```

Comment:
I don't think you need to check if the mem is really dead. Adding it to the worklist does not hurt. Would this work as well?

```
if (can_reshape) {
  phase->is_IterGVN()->_worklist.push(mem):
}
```